src/ucp/core/ucp_ep.inl

```diff
@@ -232,6 +232,7 @@ static inline void ucp_ep_flush_state_reset(ucp_ep_h ep)
                ((flush_state->send_sn == 0) &&
                 (flush_state->cmpl_sn == 0) &&
                 ucs_hlist_is_empty(&flush_state->reqs)));
+    ucs_assert(ep->flags & UCP_EP_FLAG_REMOTE_ID);
```

Comment:
why need this assertion?

Reply:
just to make sure the we reset flush state after updating a remote ID
so, we can rely on the flag

I think better to make updating remote ID dependant on flush state or matching context flags - done it

Reply:
i think better not introduce such dependency in this PR. i can require more substantial refactoring

Reply:
> i think better not introduce such dependency in this PR, since it can require more substantial refactoring

I have already done it :)
just need to change the sequence